File: app/src/main/java/com/shalzz/attendance/data/local/Converters.java
package com.shalzz.attendance.data.local;

import com.shalzz.attendance.wrapper.DateHelper;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.room.TypeConverter;

public class Converters {

    @NonNull
    @TypeConverter
    public Date fromString(String databaseValue) {
        return DateHelper.parseDate(databaseValue);
    }

    @NonNull
    @TypeConverter
    public String dateToString(@NonNull Date value) {
        return DateHelper.toTechnicalFormat(value);
    }

    @NonNull
    @TypeConverter
    public static List<Date> fromDateListString(@NonNull String databaseValue) {
        List<Date> dates = new ArrayList<>();
        for(String date : databaseValue.split(",")) {
            dates.add(DateHelper.parseDate(date));
        }
        return dates;
    }

    @NonNull
    @TypeConverter
    public static String dateListToString(@NonNull List<Date> value) {
        String SEPARATOR = ",";
        StringBuilder csvBuilder = new StringBuilder();
        for(Date date : value){
            csvBuilder.append(DateHelper.toTechnicalFormat(date));
            csvBuilder.append(SEPARATOR);
        }

        String csv = csvBuilder.toString();
        //Remove last comma
        csv = csv.substring(0, csv.length() - SEPARATOR.length());

        return csv;
    }
}

File: app/src/main/java/com/shalzz/attendance/data/local/Converters.kt
/*
 * Copyright (c) 2013-2018 Shaleen Jain <shaleen.jain95@gmail.com>
 *
 * This file is part of College Academics.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package com.shalzz.attendance.data.local

import com.shalzz.attendance.wrapper.DateHelper

import java.util.ArrayList
import java.util.Date
import androidx.room.TypeConverter

class Converters {

    @TypeConverter
    fun fromString(databaseValue: String): Date {
        return DateHelper.parseDate(databaseValue)
    }

    @TypeConverter
    fun dateToString(value: Date): String {
        return DateHelper.toTechnicalFormat(value)
    }

    @TypeConverter
    fun fromDateListString(databaseValue: String): List<Date> {
        val dates = ArrayList<Date>()
        for (date in databaseValue.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()) {
            dates.add(DateHelper.parseDate(date))
        }
        return dates
    }

    @TypeConverter
    fun dateListToString(value: List<Date>): String {
        val SEPARATOR = ","
        val csvBuilder = StringBuilder()
        for (date in value) {
            csvBuilder.append(DateHelper.toTechnicalFormat(date))
            csvBuilder.append(SEPARATOR)
        }

        var csv = csvBuilder.toString()
        //Remove last comma
        return if (csv.isEmpty()) csv else csv.substring(0, csv.length - SEPARATOR.length)
    }
}
